File: src/types/traditionArtifacts/sub/AnimalShapePath.ts
import * as DB from "tsondb/schema/dsl"
import { NestedTranslationMap } from "../../Locale.js"

export const AnimalShapePath = DB.Entity(import.meta.url, {
  name: "AnimalShapePath",
  namePlural: "AnimalShapePaths",
  type: () =>
    DB.Object({
      translations: NestedTranslationMap(
        DB.Required,
        "AnimalShapePath",
        DB.Object({
          name: DB.Required({
            comment: "The animal shape path’s name.",
            type: DB.String({ minLength: 1 }),
          }),
        }),
      ),
    }),
  instanceDisplayName: {},
})

File: Sources/OptolithDatabaseSchema/Types/TraditionArtifacts/Sub/AnimalShapePath.swift
import FileDB

@Model
public struct AnimalShapePath {

    /// All translations for the entry, identified by IETF language tag (BCP47).
    @Relationship(Locale.self)
    let translations: [String: Translation]

    @Embedded
    struct Translation { // AnimalShapePathTranslation

        /// The animal shape path’s name.
        @MinLength(1)
        let name: String
    }
}
